### DISCUSSION TITLE
Full styling documentation

### DISCUSSION THREAD

**Discussion by Pyrdacor** [AvaloniaUI/Avalonia#7545] [Q&A] [answered]
I find it quite hard to make even the easiest styling things work and the styling documentation is rather short and only shows some basic selectors and setters. The resources on the web are rare as well. Is there some more detailed documentation somewhere? I mean something which lists all the possibilities for styling?

The styling difficulties made me almost drop Avalonia completely. I just want to change the appearance of a TreeView but I can't even colorize the top-most tree view item whenever it is hovered. I mean I managed to do so after hours (which is bad) but now it is only colored when hovering the sub-items but not when it itself is hovered or selected.

I can't find anything about tree view styling and the things I found on the web are not working at all.  For example this (https://github.com/AvaloniaUI/Avalonia/issues/4847) is not working as it doesn't even compile or won't change the appearance. All the template stuff etc is very unintuitive and I don't even know which properties can be used for which controls and at which location. :(

Is there the possibility to avoid XAML/AXAML editing for such things? I really liked the WinForms editor with simple properties.

Why is everything getting more complicated? UI creation should be simple and intuitive and should not require to read books or spend hours to read documentation.

Maybe it's just about missing documentation but I spend two days now for a very simple thing and this is very frustrating. I mean I can decode old m68k assembler code easily but can't even create a simple UI with such a powerful UI framework? Something is just wrong here.

Maybe I just missed something but in general I can find some good resources with google. Not so with Avalonia unfortunately. Can you point me in the right direction? Are there better resources somewhere?

Thanks in advance

**Comment by ahopper** [comment, marked as answer]
The general approach to styling existing controls is to look at the src for the theme you are using and copy the bits you want to change e.g. https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Fluent/Controls/TreeViewItem.xaml , it is work checking that you are using the branch of the source for the version of Avalonia you are using.

**Reply by Pyrdacor** [reply]
After some time I managed to do so. But I think there should be at least some short documentation about this.

**Reply by ahopper** [reply]
You are right, and in re reading your original question and my reply I apologise for being so terse, the documentation is getting better all the time but very much benefits from input from those not so tied into its development so I'm sure any contributions to the docs would be very appreciated.

**Reply by Pyrdacor** [reply]
You are right. I really start liking the project. Maybe I'll find the time to contribute some kind of documentation while I learn all the stuff. ;)
